tests/templates/kuttl/smoke-nodeport/01-create-nginx-statefulset.yaml

```diff
@@ -16,10 +16,10 @@ spec:
     spec:
       containers:
         - name: nginx
-          image: nginx:1.23.1
+          image: nginxinc/nginx-unprivileged:1.25.2
```

Comment:
Is this a restriction we've run into on OpenShift or somewhere? `nginxinc` does indeed _seem_ to be operated by the real F5/Nginx, but that's still a fair bit more work to validate. I also don't know whether they are excluded from Docker Hub's pull rate limits?

Reply:
> Is this a restriction we've run into on OpenShift or somewhere?

Yes, on OpenShift, running the nginx image fails with the following error:

```
nginx: [emerg] mkdir() "/var/cache/nginx/client_temp" failed (13: Permission denied)
```

The nginx-unprivileged image fixes this and other issues on OpenShift.

> nginxinc does indeed seem to be operated by the real F5/Nginx, but that's still a fair bit more work to validate.

nginxinc is labeled as "Verified Publisher" on DockerHub (https://hub.docker.com/u/nginxinc). According to  nginxinc/docker-nginx-unprivileged#19, "this image is as "official" as an image without the official images tag can be". But I understand your concern.

> I also don't know whether they are excluded from Docker Hub's pull rate limits?

They are excluded according to the description of the "Verified Publisher" label:

> High-quality images from publishers verified by Docker. These products are published and maintained directly by a commercial entity. These images are not subject to rate limiting.

But if you want, we can mirror it in our repository.

Reply:
> Yes, on OpenShift, running the nginx image fails with the following error:

Ah, I see. Yeah that seems like a fair reason to replace it then.

> nginxinc is labeled as "Verified Publisher" on DockerHub (https://hub.docker.com/u/nginxinc). According to https://github.com/nginxinc/docker-nginx-unprivileged/issues/19, "this image is as "official" as an image without the official images tag can be". But I understand your concern.

I know, but so are MSSQL's images. I wouldn't consider it a mark of quality or reproducibility in itself... :p

> They are excluded according to the description of the "Verified Publisher" label:

Alright.

> But if you want, we can mirror it in our repository.

I'm fine skipping this for a test-only image, given everything else.